🔨 chore: Add Codecov Bundle Analysis

katia-sentry opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [ ] 📝 docs
- [x] 🔨 chore

#### 🔀 变更说明 | Description of Change

- Add @codecov/nextjs-webpack-plugin for bundle size tracking
- Configure bundle analysis settings in codecov.yml
- Update CI workflow to include bundle analysis step
- Set bundle change threshold to 10KB

#### 📝 补充信息 | Additional Information

Add Codecov Bundle Analysis integration

Commit:
Add Codecov Bundle Analysis integration

- Add @codecov/webpack-plugin for bundle size tracking
- Configure bundle analysis settings in codecov.yml
- Update CI workflow to include bundle analysis step
- Set bundle change threshold to 10KB

arvinxx commented:
I think it may not a good practice at this time as nextjs will mark turbopack prod stable soon? Do Codecov have an alternative in turbopack?

fyi: https://areweturboyet.com

Commit:
Use nextjs-webpack-plugin

katia-sentry commented:
@arvinxx thanks for the feedback - turbopack doesn’t support webpack plugins yet and we have opened an issue to track it https://github.com/codecov/codecov-javascript-bundler-plugins/issues/251. We can update this once support is available. It’s up to the team whether to merge now or revisit later - happy to adjust as needed!

arvinxx commented:
@katia-sentry can we have a separate build test workflow？ As the build will need 10mins+ to finish

Commit:
Seperate workflow

Move the Build with Bundle Analysis step to its own GitHub Actions workflow file.

katia-sentry commented:
@arvinxx I've moved the bundle analysis to a separate workflow (bundle-analysis.yml) so it no longer affects the test workflow. Let me know if you have any feedback!

katia-sentry commented:
Hey @arvinxx, any updates here?

arvinxx commented on .github/workflows/test.yml:
please revert this update

arvinxx reviewed (commented):
LGTM for others part

katia-sentry committed:
Revert line additon change

katia-sentry committed:
Merge branch 'main' into feat/codecov-bundle-analysis

katia-sentry commented:
@arvinxx  let me know if you need me to make anymore changes. Thank you!

arvinxx committed:
Merge branch 'main' into feat/codecov-bundle-analysis

arvinxx commented:
Thanks for your contribution!
